src/t_list.c

```diff
@@ -363,26 +363,30 @@ void lsetCommand(client *c) {
     }
 }
 
+void listElementsRemoved(client *c, int where, robj *o, robj *value) {
+    char *event = (where == LIST_HEAD) ? "lpop" : "rpop";
+
+    decrRefCount(value);
+    notifyKeyspaceEvent(NOTIFY_LIST, event, c->argv[1], c->db->id);
+    if (listTypeLength(o) == 0) {
+        notifyKeyspaceEvent(NOTIFY_GENERIC, "del", c->argv[1], c->db->id);
+        dbDelete(c->db, c->argv[1]);
+    }
+    signalModifiedKey(c, c->db, c->argv[1]);
```

Comment:
this function must not assume that argv[1] is the key name, it should get it as an argument from the caller.